vault/token_store_test.go

```diff
@@ -832,6 +832,35 @@ func TestTokenStore_HandleRequest_CreateToken_NonRoot_InvalidSubset(t *testing.T
 	}
 }
 
+func TestTokenStore_HandleRequest_CreateToken_NonRoot_RootChild(t *testing.T) {
+	_, ts, _, root := TestCoreWithTokenStore(t)
+	testMakeToken(t, ts, root, "sudoClient", "", []string{"create", "update", "sudo"})
+
+	req := logical.TestRequest(t, logical.UpdateOperation, "create")
+	req.ClientToken = "sudoClient"
+	req.Data["policies"] = []string{"root", "create"}
+
+	resp, err := ts.HandleRequest(req)
+	if err != logical.ErrInvalidRequest {
+		t.Fatalf("err: %v %v", err, resp)
+	}
+	if resp.Data["error"] != "root tokens may not be created without parent token being root" {
```

Comment:
`resp.Data` is a `map[string]interface{}`; as a result what comes out isn't a string you can compare against, but an `interface{}`, so you need to type-assert it. `resp.Data["error"].(string)` should work.